### PR TITLE
MAID-1391 Allow reusing existing UDP sockets for bind and connect

### PR DESCRIPTION
**Pull request by madadam**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/rust-utp/4)

### CODE REVIEW

File: src/socket.rs [hunk 1]
```diff
@@ -1294,6 +1318,41 @@ mod test {
...
+            // Check proper difference in client's send connection id and receive connection id
+            assert_eq!(client.sender_connection_id, client.receiver_connection_id + 1);
+            assert_eq!(client.connected_to,
+                       server_addr.to_socket_addrs().unwrap().next().unwrap());
+            iotry!(client.close());
+            drop(client);
+        });
+
+        let mut buf = [0u8; BUF_SIZE];
+        match server.recv_from(&mut buf) {
```
Comment (vinipsmaker) [inline src/socket.rs]: what's the purpose of this `match`? Why not just:

``` rust
println!("{:?}", server.recv_from(&mut buf));
```

But anyways, the `println!` doesn't test anything, so I assume you used it to "debug" something and forgot to take it away afterwards.

**Comment by vinipsmaker** [comment]
The rest of the PR seems okay and I'm looking forward to merge it.

**Comment by vinipsmaker** [comment]
Hey @madadam, could you give us your bitcoin address?

**Comment by madadam** [comment]
Sure: 1J4t5y4XJisLPv7tvCJcpho9GweM9LRoEx

**Comment by NickLambert** [comment]
0.32382 btc on the way. Thanks for your help!

**Pull request merged**